pkg/apis/discovery/types.go

```diff
@@ -114,8 +114,22 @@ type EndpointConditions struct {
 	// ready indicates that this endpoint is prepared to receive traffic,
 	// according to whatever system is managing the endpoint. A nil value
 	// indicates an unknown state. In most cases consumers should interpret this
-	// unknown state as ready.
+	// unknown state as ready. For compatibility reasons, ready should never be
+	// "true" for terminating endpoints.
 	Ready *bool
+
+	// accepting is identical to ready except it is set regardless of the terminating
+	// state of endpoints. A ready endpoint that is terminating should set this condition
+	// to true. A nil vaue indicates an unknown state. Consumers should interepret this
+	// unknown state to mean the endpoint is accepting traffic.
```

Comment:
I think it would be good to address some potential points of confusion here. I think we should update this to cover the following:
- This field may be set less often than `ready`. When it is not set, consumers should defer to `ready`. I'm not sure if that's accurate, just seems like it could be better than assuming `nil == true` here.
- It could be helpful to clarify that this should be set to true if `service.spec.publishNotReadyAddresses` is set to true. Maybe we don't actually want to carry that through to this field though? This could be our opportunity to drop that once and for all...

Reply:
> This field may be set less often than ready. When it is not set, consumers should defer to ready. I'm not sure if that's accurate, just seems like it could be better than assuming nil == true here.

Agreed that we should document users should check `Ready` if `Accepting is nil.

Reply:
> It could be helpful to clarify that this should be set to true if service.spec.publishNotReadyAddresses is set to true. Maybe we don't actually want to carry that through to this field though? This could be our opportunity to drop that once and for all...

I'm in favor of dropping `publishNotReadyAddresses` in the context of EndpointSlice. This behavior doesn't change during alpha though so I was hoping we can come to a conclusion on this before this feature goes Beta and all endpoints regardless of their condition is added to EndpointSlice.

Reply:
I think it's fair to document that publishNotReadyAddresses is ignored for EndpointSlice

Reply:
I missed the later comment.  OK.  So:

if publishNotReadyAddresses - all endpoints are always Ready.  We could still make Serving and Terminating be sane?
else 
  - Ready = Serving && !Terminating
  - Serving means pod is Ready (terminating or not) 
  - Terminating means pod is deleting (ready or not)

Yes?

Reply:
Yeah that makes sense.